File: Week01/Decimal_to_Binary/Division_by_Two.cpp
#include<iostream>
#include<algorithm>
using namespace std;

string decToBinary(int n) {
    string bin = "";
    while (n > 0) {
        int bit = n % 2;
        bin.push_back('0' + bit);
        n /= 2;
    }

    reverse(bin.begin(), bin.end());
    return bin;
}


int main() {
    int n = 12;
    cout << decToBinary(n);
    return 0;
}

File: Week01/Decimal_to_Binary/Division_by_Two.cs
using static System.Console;
using System.Collections.Generic;

class Division_by_Two
{
    static string decToBinary(int n)
    {
        List<char> bin = new List<char>();

        while (n > 0)
        {
            int bit = n % 2;
            bin.Add((char)('0' + bit));
            n /= 2;
        }

        bin.Reverse();
        return new string(bin.ToArray());
    }

    static void Main()
    {
        int n = 12;
        WriteLine(decToBinary(n));
    }
}